apis/apiextensions/v1/composition_patches.go

```diff
@@ -276,7 +284,7 @@ func (c *Patch) applyCombineFromVariablesPatch(from, to runtime.Object) error {
 		return err
 	}
 
-	return patchFieldValueToObject(*c.ToFieldPath, out, to)
+	return patchFieldValueToObject(*c.ToFieldPath, out, to, nil)
```

Comment:
I imagine we might want to pass merge options here too at some point; in fact it's a little misleading that `mergeOptions` are currently ignored by patches of `type: CombineFromComposite` and `type: CombineToComposite`. I don't think it makes sense to implement now though, given that AFAICT it's not actually possible for the combine patch types to produce a map or array.

Reply:
Agreed that passing them in here will make sense once the combine types support arrays/maps.  As you pointed out, I didn't implement it with this PR because of the fact that it's basically a no-op at this point.